### PR TITLE
added a configurable WORKOS_COOKIE_NAME optional env var

### PR DESCRIPTION
**Pull request by mattbf**
Fixes issue #58 

Right now, if you set a cookie on your backend and try to use AuthKit with nextjs on your frontend, it will not work unless you set the cookieName to "wos-session".

This would allow users who are integrating with workOS in a full stack application to customize the cookieName used. For example, in a nodejs backend: 
```ts
res.cookie("myapp-session", encryptedSession, {
  path: "/",
  httpOnly: true,
  secure: true,
  sameSite: "lax",
});
```

Then you would set the optional env var `WORKOS_COOKIE_NAME` on the frontend and AuthKit would automatically and properly consume the cookie 🍪 😄

### CODE REVIEW

**Comment by PaulAsjes** [comment]
Thanks for this! The code looks good but I'm going to hold on merging this till we figure out a solution for https://github.com/workos/authkit-nextjs/issues/24 and https://github.com/workos/authkit-nextjs/issues/45.

File: src/session.ts [hunk 1]
```diff
@@ -5,14 +5,14 @@ import { jwtVerify, createRemoteJWKSet, decodeJwt } from 'jose';
...
+import { WORKOS_CLIENT_ID, WORKOS_COOKIE_PASSWORD, WORKOS_REDIRECT_URI, WORKOS_COOKIE_NAME } from './env-variables.js';
 import { getAuthorizationUrl } from './get-authorization-url.js';
 import { AccessToken, AuthkitMiddlewareAuth, NoUserInfo, Session, UserInfo } from './interfaces.js';
 
 import { parse, tokensToRegexp } from 'path-to-regexp';
 
-const sessionHeaderName = 'x-workos-session';
-const middlewareHeaderName = 'x-workos-middleware';
+const sessionHeaderName = !!WORKOS_COOKIE_NAME ? `x-${WORKOS_COOKIE_NAME}` : 'x-workos-session';
+const middlewareHeaderName = !!WORKOS_COOKIE_NAME ? `x-${WORKOS_COOKIE_NAME}-middleware` : 'x-workos-middleware';
```
Comment (PaulAsjes) [inline src/session.ts:15]: I don't think these need to be configurable. These are specifically used for some behind-the-scenes work (mostly to improve the DX of working with Next.js middleware) and aren't intended for use outside of this library.

**Review by PaulAsjes — CHANGES_REQUESTED** [review]

File: src/session.ts (hunk 1 above)
Comment (mattbf) [reply src/session.ts]: Roger! I removed these edits

**Comment by PaulAsjes** [comment]
Released a variation of this in 0.11.0.

**Comment by mattbf** [comment]
Cheers, thanks @PaulAsjes!

**Pull request closed without merging**